### PR TITLE
adding delete_node

### PR DESCRIPTION
**Pull request by ADSmarcossobral**

### CODE REVIEW

File: forest/binary_search_tree.h [hunk 1]
```diff
@@ -255,6 +255,54 @@ namespace forest {
                                 return false;
                         }
                 }
+                /**
+                 * @brief Deletes a node with a specific key
+                 * @param key The key for node deletetion
+                 * @return true to success, otherwise false
+                 */
+                bool delete_node(key_t key) {
+                        const splay_tree_node <key_t, value_t> *x = search(key);
```
Comment (xorz57) [inline forest/binary_search_tree.h]: Why do you use a splay_tree_node here?

Also, search(...) returns a pointer to a const object. You can't modify that object.

**Comment by ADSmarcossobral** [comment]
I checked the errors, I will correct them.

**Comment by xorz57** [comment]
@ADSmarcossobral Let me know when you are ready! 📬

**Comment by ADSmarcossobral** [comment]
I'm sorry. I don't understand!

**Pull request closed without merging**